plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeColumnHandle.java

```diff
@@ -131,61 +142,84 @@ public boolean equals(Object obj)
             return false;
         }
         DeltaLakeColumnHandle other = (DeltaLakeColumnHandle) obj;
-        return Objects.equals(this.name, other.name) &&
-                Objects.equals(this.type, other.type) &&
+        return Objects.equals(this.baseColumnName, other.baseColumnName) &&
+                Objects.equals(this.baseType, other.baseType) &&
                 Objects.equals(this.fieldId, other.fieldId) &&
-                Objects.equals(this.physicalName, other.physicalName) &&
-                Objects.equals(this.physicalType, other.physicalType) &&
-                this.columnType == other.columnType;
+                Objects.equals(this.basePhysicalColumnName, other.basePhysicalColumnName) &&
+                Objects.equals(this.basePhysicalType, other.basePhysicalType) &&
+                this.columnType == other.columnType &&
+                Objects.equals(this.projectionInfo, other.projectionInfo);
+    }
+
+    @JsonIgnore
+    public String getQualifiedName()
+    {
+        return projectionInfo.map(projectionInfo -> basePhysicalColumnName + "#" + projectionInfo.getPartialName())
+                .orElse(basePhysicalColumnName);
     }
 
     public long getRetainedSizeInBytes()
     {
         // type is not accounted for as the instances are cached (by TypeRegistry) and shared
-        return INSTANCE_SIZE + estimatedSizeOf(name);
+        return INSTANCE_SIZE
+                + estimatedSizeOf(baseColumnName)
+                + (projectionInfo.isPresent() ? estimatedSizeOf(projectionInfo.get().getPartialName()) : 0);
     }
 
     @Override
     public int hashCode()
     {
-        return Objects.hash(name, type, fieldId, physicalName, physicalType, columnType);
+        return Objects.hash(baseColumnName, baseType, fieldId, basePhysicalColumnName, basePhysicalType, columnType, projectionInfo);
     }
 
     @Override
     public String toString()
     {
-        return name + ":" + type.getDisplayName() + ":" + columnType;
+        return getQualifiedName() +
+                ":" + (projectionInfo.isEmpty() ? baseType.getDisplayName() : projectionInfo.get().getType().getDisplayName()) +
+                ":" + columnType;
     }
 
     public HiveColumnHandle toHiveColumnHandle()
     {
+        return toHiveColumnHandle(basePhysicalColumnName); // this name is used for accessing Parquet files, so it should be physical name
+    }
+
+    public HiveColumnHandle toHiveColumnHandle(String baseColumnName)
```

Comment:
This method has a strange contract to me. "Make a HiveColumnHandle from this Delta column, but override the base column name". Personally I'd rather remove this from here and do inline it into where it's used in the PageSourceProvider